src/libs/antares/study/parameters.cpp

```diff
@@ -697,6 +723,35 @@ static bool SGDIntLoadFamily_P(Parameters& d, const String& key, const String& v
         return false;
     }
 
+    /* Format added in Antares 8.1 */
+    if (key.startsWith("playlist_weight_year_"))
+    {
+        String key_cpy(key);
+        uint year_in;
+        float weight_in;
+        key_cpy.replace("playlist_weight_year_", "");
+        if (!key_cpy.to<uint>(year_in))
+        {
+            logs.error() << "parameters: invalid MC year index. Got '" << key << "'";
+            return false;
+        }
+        if (!value.to<float>(weight_in))
+        {
+            logs.error() << "parameters: invalid MC year weight. Got '" << value << "'";
+            return false;
+        }
+
+        if (weight_in < 0.f)
+        {
+            logs.warning() << "parameters: invalid MC year weight.Got '" << weight_in
+                           << "'. Value not used";
+            weight_in = 1.f;
+        }
+
+        d.setYearWeight(year_in, weight_in);
+        return true;
```

Comment:
Could we use a kind of a map ?
`{1: 2.3 , 2 : 4.5}`

Reply:
we could, but it's harder to parse and possibly less human-readable.